.github/promote-images.yml

```diff
@@ -1,43 +1,43 @@
 # Sync all the images that are tagged using date time stamp
 # format +%Y-%m-%d-%H-%M-%S e.g. 2022-02-22-10-10-54.
 # Use simple regex for readability. Since we tag all images with year
-# We use a regex rule: Any tag that starts with '2024'.
+# We use a regex rule: Any tag that starts with 'CURRENT_YEAR-CURRENT_MONTH'.
 "europe-docker.pkg.dev/gitpod-artifacts/docker-dev":
   images-by-tag-regex:
-    workspace-base: "2024.*"
-    workspace-c: "2024.*"
-    workspace-clojure: "2024.*"
-    workspace-elixir: "2024.*"
-    workspace-full: "2024.*"
-    workspace-full-vnc: "2024.*"
-    workspace-go: "2024.*"
-    workspace-nix: "2024.*"
-    workspace-node: "2024.*"
-    workspace-node-lts: "2024.*"
-    workspace-node-18: "2024.*"
-    workspace-node-20: "2024.*"
-    workspace-node-22: "2024.*"
-    workspace-python: "2024.*"
-    workspace-python-3.8: "2024.*"
-    workspace-python-3.9: "2024.*"
-    workspace-python-3.10: "2024.*"
-    workspace-python-3.11: "2024.*"
-    workspace-python-3.12: "2024.*"
-    workspace-ruby-3: "2024.*"
-    workspace-ruby-3.1: "2024.*"
-    workspace-ruby-3.2: "2024.*"
-    workspace-ruby-3.3: "2024.*"
-    workspace-rust: "2024.*"
-    workspace-dotnet: "2024.*"
-    workspace-dotnet-6: "2024.*"
-    workspace-dotnet-8: "2024.*"
-    workspace-postgres: "2024.*"
-    workspace-mysql: "2024.*"
-    workspace-mongodb: "2024.*"
-    workspace-java-11: "2024.*"
-    workspace-java-17: "2024.*"
-    workspace-java-21: "2024.*"
-    workspace-yugabytedb: "2024.*"
-    workspace-yugabytedb-preview: "2024.*"
-    workspace-gitpod-dev: "2024.*"
-    workspace-bun: "2024.*"
+    workspace-base: "CURRENT_YEAR-CURRENT_MONTH.*"
```

Comment:
nice!

I wonder if there are any implications on images potentially not being promoted from the end of a month? If the sync job failed to sync on the 31st, and you re-ran the following day, it would most likely not sync that release. Sounds acceptable though.

Reply:
I agree. It would be included "the next time" we update the base image, which happens regularly enough, I think.